vm/src/hint_processor/builtin_hint_processor/bootloader_hints.rs

```diff
@@ -272,4 +372,158 @@ mod tests {
+        );
+    }
+
+    #[test]
+    fn test_save_packed_ouputs() {
+        use crate::hint_processor::builtin_hint_processor::hint_code::BOOTLOADER_SAVE_PACKED_OUTPUTS;
+
+        let packed_outputs = vec![
+            PackedOutput {},
+            PackedOutput {},
+            PackedOutput {},
+        ];
+
+        let bootloader_input = BootloaderInput {
+            simple_bootloader_input: SimpleBootloaderInput { fact_topologies_path: None, single_page: false },
+            bootloader_config: BootloaderConfig {
+                simple_bootloader_program_hash: ProgramHash(42u64),
+                supported_cairo_verifier_program_hashes: Default::default(),
+            },
+            packed_outputs: packed_outputs.clone(), 
+        };
+
+        let mut vm = vm!();
+        let mut exec_scopes = ExecutionScopes::new();
+
+        exec_scopes.insert_box("bootloader_input", Box::new(bootloader_input.clone()));
+
+        let hint_data =
+            HintProcessorData::new_default(String::from(BOOTLOADER_SAVE_PACKED_OUTPUTS), HashMap::new());
+        let mut hint_processor = BuiltinHintProcessor::new_empty();
+        assert_matches!(
+            hint_processor.execute_hint(
+                &mut vm,
+                &mut exec_scopes,
+                &any_box!(hint_data),
+                &HashMap::new(),
+            ),
+            Ok(())
+        );
+
+        let saved_packed_outputs = exec_scopes.get::<Vec<PackedOutput>>("packed_outputs");
+        assert_matches!(
+            saved_packed_outputs,
+            Ok(ref x) if x == &packed_outputs
+        );
+
+        assert_eq!(saved_packed_outputs.expect("asserted Ok above, qed").len(), 3);
+    }
+
+    #[test]
+    fn test_set_packed_output_to_subtasks_hint() {
+        use crate::hint_processor::builtin_hint_processor::hint_code::BOOTLOADER_SET_PACKED_OUTPUT_TO_SUBTASKS;
+        use felt::Felt252;
+
+        let mut vm = vm!();
+        let mut exec_scopes = ExecutionScopes::new();
+
+        exec_scopes.insert_box("packed_output", Box::new(Felt252::from(42)));
+
+        let hint_data =
+            HintProcessorData::new_default(String::from(BOOTLOADER_SET_PACKED_OUTPUT_TO_SUBTASKS), HashMap::new());
+        let mut hint_processor = BuiltinHintProcessor::new_empty();
+        assert_matches!(
+            hint_processor.execute_hint(
+                &mut vm,
+                &mut exec_scopes,
+                &any_box!(hint_data),
+                &HashMap::new(),
+            ),
+            Ok(())
+        );
+
+        let packed_outputs = exec_scopes.get::<Felt252>("packed_outputs");
+        assert_matches!(
+            packed_outputs,
+            Ok(x) if x == Felt252::from(42)
+        );
+    }
+
+    #[test]
+    fn test_guess_pre_image_of_subtasks_output_hash_hint() {
+        use crate::hint_processor::builtin_hint_processor::hint_code::BOOTLOADER_GUESS_PRE_IMAGE_OF_SUBTASKS_OUTPUT_HASH;
+
+        let mut vm = vm!();
```

Comment:
```suggestion
        let mut vm = vm!();
        add_segments!(vm, 2);
        vm.run_context.fp = 2;
        let ids_data = ids_data!["nested_subtasks_output_len", "nested_subtasks_output"];

```

This makes the test run properly. This:
* declares the two variables in the `ids_data` dictionary
* allocates memory segments to store the variables
* sets FP to 2, as it should point to the first free address in the segment.